### PR TITLE
feat: app shell with navigation, theming, and dashboard

### LINKED ISSUE #12: App shell and navigation
## Summary

Build the Angular Material app shell with sidebar navigation, theming, and responsive layout that serves as the container for all feature modules. This is the visual foundation — every feature renders inside this shell. Includes a home dashboard as the landing page.

## Features / Scope

| Feature | Description |
|---------|-------------|
| App shell | Material Design 3 shell with top toolbar and collapsible sidebar |
| Navigation | Sidebar with links to: Home/Dashboard, Design, Catalog, Metrics, Knowledge Base. Active route highlighted. |
| Home dashboard | Landing page with recent designs, quick-start actions, and system overview |
| Theming | Custom fabrik theme (deep blue/slate primary, teal accent) with easy re-theming via CSS custom properties |
| Dark mode | Toggle between light and dark themes, persisted to localStorage |
| Lazy routing | Feature modules lazy-loaded via Angular router with loading indicators |
| Responsive layout | Collapsible sidebar on smaller screens, mobile-friendly breakpoints |

## Architecture / Design Notes

- **Module location**: App shell component in `frontend/src/app/core/`, home dashboard in `frontend/src/app/features/dashboard/`
- **Theming**: Define theme using Angular Material's M3 theming system in `src/styles.scss`. All custom colors use CSS custom properties (e.g., `--fabrik-primary`, `--fabrik-accent`) so re-theming requires changing only the property definitions — no component-level style changes.
- **Color palette**: Deep blue/slate primary (`#1e293b` family), teal accent (`#0d9488` family). Professional, infrastructure-appropriate.
- **Dark mode**: Theme toggle in toolbar. Preference stored in `localStorage` and applied on load via a `data-theme` attribute on `<body>`. Angular Material's built-in dark mode support handles component theming.
- **Routing**: Feature modules register as lazy-loaded routes in `frontend/src/app/app.routes.ts`. Each feature exports a `Routes` array from its own `*.routes.ts` file. The shell doesn't import feature modules directly.
- **Navigation state**: Sidebar highlights the active route using Angular Router's `routerLinkActive`. Sidebar collapse state persisted to `localStorage`.
- **Layout**: Angular Material `mat-sidenav-container` with `mat-sidenav` (sidebar) and `mat-sidenav-content` (main area). Toolbar uses `mat-toolbar`.

## Integration Points

- **Every feature module**: Renders inside the shell's `<router-outlet>`. Features register their routes in `app.routes.ts` as lazy-loaded children.
- **Knowledge base panel (#11)**: The shell provides the container for the slide-out help panel. The `KnowledgePanelService` in `core/` manages panel state; the shell template includes the panel outlet.
- **Home dashboard**: Consumes the Design CRUD API (#1) to show recent designs. Quick-start actions link to the Design and Catalog features.
- **Route convention**: Each feature owns a top-level route (`/design`, `/catalog`, `/metrics`, `/knowledge`). The shell's sidebar is the only place these routes are enumerated.

## Acceptance Criteria

- [ ] Material Design 3 app shell with top toolbar and sidebar navigation
- [ ] Sidebar navigation with links to: Home, Design, Catalog, Metrics, Knowledge Base
- [ ] Active route highlighted in sidebar
- [ ] Home dashboard as default route (`/`) with recent designs list and quick-start actions
- [ ] Responsive layout: sidebar collapses to hamburger menu below 768px breakpoint
- [ ] Sidebar collapse state persisted to `localStorage`
- [ ] Custom fabrik theme: deep blue/slate primary, teal accent, defined via CSS custom properties for easy re-theming
- [ ] Dark mode toggle in toolbar, preference persisted to `localStorage`, applied on page load without flash
- [ ] Lazy-loaded feature module routing with placeholder routes for Design, Catalog, Metrics, Knowledge
- [ ] Loading indicator during route transitions (progress bar or spinner)
- [ ] Route configuration in `frontend/src/app/app.routes.ts`
- [ ] App shell component in `frontend/src/app/core/`
- [ ] Dashboard component in `frontend/src/app/features/dashboard/`
- [ ] Unit tests for app shell component (renders toolbar, sidebar, router outlet)
- [ ] Unit tests for theme toggle (switches theme, persists preference, loads saved preference)
- [ ] Unit tests for sidebar (responsive collapse, active route highlighting, state persistence)
- [ ] Unit tests for dashboard component
- [ ] E2e test: navigate between all routes, verify sidebar active state, toggle dark mode
- [ ] Tests pass, build succeeds

## Edge Cases

- Deep-link directly to `/catalog` → app shell renders with sidebar, navigates to catalog route, sidebar highlights Catalog
- Browser back/forward → navigation state updates correctly, sidebar active state follows
- No saved theme preference → default to light mode, respect `prefers-color-scheme` media query
- Dark mode toggle → no flash of wrong theme on page reload (apply theme before Angular bootstraps)
- All feature routes are placeholder stubs initially → each renders a "Coming soon" message until the feature is implemented
- Very narrow viewport (< 480px) → sidebar fully hidden, accessible via hamburger menu only
- `localStorage` unavailable (private browsing edge case) → graceful fallback, no errors, just don't persist

## Documentation

- [ ] Theming guide in `CLAUDE.md`: how to change colors (CSS custom properties), how dark mode works
- [ ] Routing conventions in `CLAUDE.md`: how to add a new feature route (lazy-loaded pattern)
- [ ] Sidebar navigation: how to add a new nav item

## Dependencies

None — can be built in parallel with other features. Placeholder routes will be replaced as features are implemented.

### PR DESCRIPTION
**Pull request by rnwolfe**
## Summary
- Angular Material 3 app shell with top toolbar and collapsible sidebar (`core/shell/`)
- Sidebar navigation to Home, Design, Catalog, Metrics, Knowledge Base with active route highlighting
- Home dashboard at `/` with recent designs list (from `GET /api/designs`) and quick-start actions (`features/dashboard/`)
- Custom fabrik theme using M3 CSS variables (teal primary `#0d9488`, deep blue/slate surface `#1e293b`)
- Dark mode toggle in toolbar, persisted to `localStorage`, applied before Angular bootstraps via inline script in `index.html` (no flash)
- Lazy-loaded routes for Design, Catalog, Metrics, Knowledge with "Coming soon" stubs
- Loading indicator (`mat-progress-bar`) during route transitions
- Responsive sidebar collapse at 768px with hamburger menu; collapse state persisted to `localStorage`
- `KnowledgePanelService` in core for future knowledge base slide-out panel
- `@angular/animations` added as required dependency for Material animations

## Acceptance Criteria
- [x] Material Design 3 app shell with toolbar and sidebar
- [x] Sidebar links to: Home, Design, Catalog, Metrics, Knowledge Base (active route highlighted)
- [x] Home dashboard at `/` with recent designs list and quick-start actions
- [x] Responsive sidebar collapse at 768px with hamburger menu
- [x] Sidebar collapse state persisted to localStorage
- [x] Custom fabrik theme via CSS custom properties
- [x] Dark mode toggle, persisted, no flash on reload
- [x] Lazy-loaded routes for Design, Catalog, Metrics, Knowledge (placeholder "Coming soon" until implemented)
- [x] Loading indicator during route transitions
- [x] Route config in `frontend/src/app/app.routes.ts`
- [x] Unit tests: shell component, theme toggle, sidebar, dashboard (18 tests, all passing)
- [x] E2e test: navigate routes, verify sidebar, toggle dark mode (`frontend/e2e/app-shell.spec.ts`)
- [x] Tests pass, build succeeds

## Test plan
- [x] `make test` — all 18 unit tests pass (4 test files)
- [x] `make build` — frontend builds successfully with lazy chunks for each feature route
- [ ] E2e tests run against live server (require `ng serve` + backend)

Closes #12

🤖 Generated with Claude Code

### CODE REVIEW

File: frontend/package.json [hunk 1]
```diff
@@ -20,6 +21,7 @@
     "@angular/material": "^21.2.4",
     "@angular/platform-browser": "^21.2.0",
     "@angular/router": "^21.2.0",
+    "@testing-library/user-event": "^14.6.1",
     "rxjs": "~7.8.0",
     "tslib": "^2.3.0"
```
[the remark this answers is not in the document]
Comment (rnwolfe) [reply frontend/package.json]: Fixed in PR #16: Moved `@testing-library/user-event` from `dependencies` to `devDependencies`.
